web/modules/custom/configform/src/Form/ConfigForm.php

```diff
@@ -0,0 +1,259 @@
+
+  /**
+   * Class Property of messenger interface.
+   *
+   * @var \Drupal\Core\Messenger\MessengerInterface
+   */
+  protected $messenger;
+
+  /**
+   * The injected dependancy is assigned to the corresponding class propery.
+   *
+   * @param \Drupal\Core\Messenger\MessengerInterface $messenger
+   *   Class property of messenger interface.
+   */
+  public function __construct(MessengerInterface $messenger) {
+    $this->messenger = $messenger;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(DependencyInjectionContainerInterface $container) {
+    return new static(
+      $container->get('messenger')
+    );
+  }
+
+  /**
+   * Generating a unique form id.
+   *
+   * @return string
+   *   Unique form id
+   */
+  public function getFormId() {
+    return 'config_form';
+  }
+
+  /**
+   * This function is used to declare the type of form.
+   *
+   * @return array
+   *   It stores the config form
+   */
+  protected function getEditableConfigNames() {
+    return [
+      'config.settings',
+    ];
+  }
+
+  /**
+   * This function will faciliate building the form.
+   *
+   * @param array $form
+   *   This array contains all the fields in an associative array format.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   This variable stores the current state of the form.
+   *
+   * @return array
+   *   Array containing all the form field and data
+   */
+  public function buildForm(array $form, FormFormStateInterface $form_state) {
+    $form['error'] = [
+      '#type'      => 'markup',
+      '#markup'    => '<div id="error-message"></div>',
+    ];
+    $form['FullName'] = [
+      '#title'    => $this->t('Full Name'),
+      '#type'     => 'textfield',
+      '#required' => TRUE,
+      '#size'     => 30,
+    ];
+    $form['PhoneNumber'] = [
+      '#title'    => $this->t('Phone Number'),
+      '#type'     => 'number',
+      '#required' => TRUE,
+      '#size'     => 10,
+      '#suffix'   => '<span id="phone_error">',
+    ];
+    $form['Email'] = [
+      '#title'    => $this->t('Email'),
+      '#type'     => 'email',
+      '#required' => TRUE,
+    ];
+    $form['Gender'] = [
+      '#type'     => 'radios',
+      '#title'    => $this->t('Gender'),
+      '#options'  => [
+        'male'    => $this->t('Male'),
+        'female'  => $this->t('Female'),
+      ],
+    ];
+    $form['Subscribe'] = [
+      '#type'     => 'radios',
+      '#title'    => $this->t('Subscribe'),
+      '#required' => TRUE,
+      '#options'  => [
+        'yes'     => $this->t('Yes'),
+        'no'      => $this->t('No'),
+      ],
+      '#attributes' => [
+        'id' => 'conditional_field',
+      ],
+    ];
+    $form['subscribe_message'] = [
+      '#type'     => 'textfield',
+      '#title'    => $this->t('Why Not'),
+      '#states'   => [
+        'visible' => [
+          ':input[id="conditional_field"]' => ['value' => 'no'],
+        ],
+      ],
+    ];
+    $form['action']['submit'] = [
```

Comment:
Since this is the config form, where are we updating the configs?

Reply:
Added